final/background/repl.tex

```diff
@@ -71,36 +71,41 @@ \subsection{Advantages of REPLs}
 combined with what is called literate programming to offer notebooks or language
 playgrounds, as discussed in \cref{sec:literate-programming}.
 
-\subsection{Execution model}
+\subsection{Execution Model}
 \label{ssec:execution-model}
 
 Every programming language defines an execution model, which specifies how
-programs written in that language are executed. Amongst others, it specifies
-what an indivisable amount of work is (a \emph{compilation unit}) and in what
-order these units are executed. The implementation of an execution model is a
-compiler and/or an interpreter, often accompanied with a runtime system to
-provide features such as garbage collection.
-
-The execution model as implemented by a REPL differs only slightly to that of an
-interpreter or compiler. A difference might be as to what is considered an
-indivisable unit of work: a REPL might accept singular expressions that a
-regular interpreter or compiler might not. 
-
-A compiler or an interpreter reads the files that make up a program’s source
-code, after which it will go through the steps as outlined in
-\cref{sec:spoofax}. Usually the resulting program is executed as a whole,
-without the need for smaller blocks of execution. In contrast, when using a
-REPL, it might be desirable to be able to execute say only a function body as
-one unit of execution. However, breaking up a program into smaller blocks of
-execution might be quite challenging.
-
-Another contrast between a compiler or an interpreter and a REPL is that a REPL
-needs to dynamically maintain an environment such that each new expression can
-be evaluated within the environment of previously executed expressions. For
-every expression entered, it thus needs to apply the outlined steps again and
-update the environment it maintains in memory with the new results. This could
-result in a different order of evaluation than when say an interpreter executes
-a program as a whole.
+programs written in that language are executed. Among others, it specifies what
```

Comment:
`Amongst others` is the saying, so whilst `among` is not wrong I prefer having `amongst`.